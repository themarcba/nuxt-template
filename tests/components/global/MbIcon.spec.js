import {} from 'jest';
// Doc: https://vue-test-utils.vuejs.org/api/
import { createLocalVue, config, shallowMount } from '@vue/test-utils';
import MbIcon from '@/components/global/MbIcon.vue';
import NuxtStubs from '@/tests/nuxt-stubs';

const localVue = createLocalVue();

// Doc: https://vue-test-utils.vuejs.org/api/#config
config.showDeprecationWarnings = true;

describe('MbIcon.vue', () => {
  const wrapper = shallowMount(MbIcon, {
    localVue,
    propsData: {
      icon: 'calendar'
    },
    stubs: {
      ...NuxtStubs,
      FontAwesomeIcon: NuxtStubs.ClientOnly,
    }
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it('matches the snapshot', async () => {
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ type: 'solid' });
    expect(wrapper).toMatchSnapshot();

    await wrapper.setProps({ type: 'brand' });
    expect(wrapper).toMatchSnapshot();
  });
});
