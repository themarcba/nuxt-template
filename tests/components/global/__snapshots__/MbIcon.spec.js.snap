// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MbIcon.vue matches the snapshot 1`] = `<div icon="far,calendar"></div>`;

exports[`MbIcon.vue matches the snapshot 2`] = `<div icon="fas,calendar"></div>`;

exports[`MbIcon.vue matches the snapshot 3`] = `<div icon="fab,calendar"></div>`;
